### PR TITLE
Adding functions to access MOI attributes in nice way

### PR DESCRIPTION
**Pull request by yashcodes**
This PR adds functions to access MOI attributes `MOI.SimplexIterations`, `MOI.BarrierIterations`, `MOI.NodeCount`.
Solves [#2158](https://github.com/JuliaOpt/JuMP.jl/issues/2158)

### CODE REVIEW

File: docs/src/solutions.md [hunk 1]
```diff
@@ -132,8 +132,10 @@ end
 
 [MathOptInterface](https://www.juliaopt.org/MathOptInterface.jl/v0.9.10/) defines a large
 number of model attributes that can be queried. Examples include
-[`MOI.RelativeGap`](https://www.juliaopt.org/MathOptInterface.jl/v0.9.10/apireference/#MathOptInterface.RelativeGap) and
-[`MOI.SimplexIterations`](https://www.juliaopt.org/MathOptInterface.jl/v0.9.10/apireference/#MathOptInterface.SimplexIterations). Some attributes can be directly accessed by getter functions like [`objective_bound`](@ref) and [`relative_gap`](@ref).
+[`MOI.RelativeGap`](https://www.juliaopt.org/MathOptInterface.jl/v0.9.10/apireference/#MathOptInterface.RelativeGap)
+[`MOI.SimplexIterations`](https://www.juliaopt.org/MathOptInterface.jl/v0.9.10/apireference/#MathOptInterface.SimplexIterations)
+[`MOI.BarrierIterations`](https://www.juliaopt.org/MathOptInterface.jl/v0.9.10/apireference/#MathOptInterface.BarrierIterations)and
```
Comment (blegat) [inline docs/src/solutions.md:137]: Missing space before `and`

File: docs/src/solutions.md [hunk 2]
```diff
@@ -132,8 +132,10 @@ end
 
 [MathOptInterface](https://www.juliaopt.org/MathOptInterface.jl/v0.9.10/) defines a large
 number of model attributes that can be queried. Examples include
-[`MOI.RelativeGap`](https://www.juliaopt.org/MathOptInterface.jl/v0.9.10/apireference/#MathOptInterface.RelativeGap) and
-[`MOI.SimplexIterations`](https://www.juliaopt.org/MathOptInterface.jl/v0.9.10/apireference/#MathOptInterface.SimplexIterations). Some attributes can be directly accessed by getter functions like [`objective_bound`](@ref) and [`relative_gap`](@ref).
+[`MOI.RelativeGap`](https://www.juliaopt.org/MathOptInterface.jl/v0.9.10/apireference/#MathOptInterface.RelativeGap)
+[`MOI.SimplexIterations`](https://www.juliaopt.org/MathOptInterface.jl/v0.9.10/apireference/#MathOptInterface.SimplexIterations)
```
Comment (blegat) [inline docs/src/solutions.md:136]: Missing commas

File: src/objective.jl [hunk 3]
```diff
@@ -8,6 +8,35 @@
 # See http://github.com/JuliaOpt/JuMP.jl
 #############################################################################
 # This file contains objective-related functions
+"""
+    simplex_iterations(model::Model)
+Return the cumulative number of simplex iterations after a call to `optimize!(model)`.
```
Comment (blegat) [inline src/objective.jl:13]: Missing empty line

File: src/objective.jl [hunk 4]
```diff
@@ -8,6 +8,35 @@
...
+Exact value depends upon implementation of MathOptInterface.SimplexIterations()
```
Comment (blegat) [inline src/objective.jl:14]: Missing backquotes surounding the MOI attributes. Also add a link to the MOI doc

**Comment by odow** [comment]
Closing as duplicate of #2201

**Pull request closed without merging**